### PR TITLE
Use hardhat to run unlock-js tests

### PR DESCRIPTION
**Pull request by clemsos**
# Description

Use hardhat to run tests instead of having to launch an eth-node in a separate container.

The speed is quite similar but ease of use is much higher as one just has to run `yarn test` - no docker needed.

### BEFORE

```
Test Suites: 5 passed, 5 total
Tests:       352 passed, 352 total
Snapshots:   0 total
Time:        78.305 s
```

### AFTER

```
Test Suites: 5 passed, 5 total
Tests:       352 passed, 352 total
Snapshots:   0 total
Time:        75.818 s
```

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-js/src/unlockService.js [hunk 1]
```diff
@@ -21,6 +21,10 @@ export default class UnlockService {
     if (!this.networks[networkId]) {
       throw new Error(`Missing config for ${networkId}`)
     }
+    // for convenience, pass directly an ethers provider in the `networks` contructor
+    if (this.networks[networkId].ethersProvider) {
+      return this.networks[networkId].ethersProvider
+    }
```
Comment (clemsos) [inline packages/unlock-js/src/unlockService.js:27]: this is a require change in the library itself, so the only blocker I can think of.

File: packages/unlock-js/src/__tests__/integration/walletServiceIntegration.test.js [hunk 2]
```diff
@@ -41,15 +37,24 @@ const versions = {
...
 
   beforeAll(async () => {
+    // deploy ERC20 and set balances
+    ERC20Address = await nodeSetup()
+
+    const [signer] = await ethers.getSigners()
+    const ethersProvider = signer.provider
+
+    // pass hardhat ethers provider
+    networks[chainId].ethersProvider = ethersProvider
```
Comment (clemsos) [inline packages/unlock-js/src/__tests__/integration/walletServiceIntegration.test.js:53]: this requires a change in unlock-js itself: the `networks` object pass to constructor now accept a `ethersProvider` key that contains an ethers js provider.

File: packages/unlock-js/jest.config.js [hunk 3]
```diff
@@ -9,6 +9,6 @@ module.exports = {
     '<rootDir>/src/__tests__/setup/',
   ],
   transform: {
-    '^.+\\.js$': 'babel-jest',
+    '^.+\\.jsx?$': require.resolve('./jest.transform.js'),
```
Comment (julien51) [inline packages/unlock-js/jest.config.js:12]: On a du jsx ici?

**Review by julien51 — APPROVED** [review]
Let's do it!

File: packages/unlock-js/jest.config.js (hunk 3 above)
Comment (clemsos) [reply packages/unlock-js/jest.config.js]: nah I think the question mark after the x in the regex makes it optional. Changed to js anyway

**Pull request merged**